librz/util/print.c

```diff
@@ -1036,39 +1097,51 @@ RZ_API void rz_print_hexdump(RzPrint *p, ut64 addr, const ut8 *buf, int len, int
 						} else {
 							a = "";
 						}
-						printfmt("%s  ; %s", a, comment);
+						rz_strbuf_appendf(sb, "%s  ; %s", a, comment);
 						free(comment);
 					}
 				}
 			}
 			if (use_align && rowbytes < inc && bytes >= rowbytes) {
 				i -= (inc - bytes);
 			}
-			print("\n");
+			rz_strbuf_append(sb, "\n");
 		}
 		rows++;
 		bytes = 0;
 		if (p && p->cfmt && *p->cfmt) {
 			if (row_have_cursor != -1) {
 				int i = 0;
-				print(" _________");
+				rz_strbuf_append(sb, " _________");
 				if (!compact) {
-					print("_");
+					rz_strbuf_append(sb, "_");
 				}
 				for (i = 0; i < row_have_cursor; i++) {
 					if (!pairs || (!compact && i % 2)) {
-						print("___");
+						rz_strbuf_append(sb, "___");
 					} else {
-						print("__");
+						rz_strbuf_append(sb, "__");
 					}
 				}
-				print("__|\n");
-				printfmt("| cmd.hexcursor = %s\n", p->cfmt);
+				rz_strbuf_append(sb, "__|\n");
+				rz_strbuf_appendf(sb, "| cmd.hexcursor = %s\n", p->cfmt);
 				p->coreb.cmdf(p->coreb.core,
 					"%s @ 0x%08" PFMT64x, p->cfmt, row_have_addr);
 			}
 		}
 	}
+	return rz_strbuf_drain(sb);
+}
+
+/// \see rz_print_hexdump_str
+RZ_API void rz_print_hexdump(RZ_NONNULL RzPrint *p, ut64 addr, RZ_NONNULL const ut8 *buf, int len, int base, int step, size_t zoomsz) {
+	rz_return_if_fail(p && buf);
+	char *str = rz_print_hexdump_str(p, addr, buf, len, base, step, zoomsz);
+	if (!str) {
+		return;
+	}
+	p->cb_printf("%s", str);
```

Comment:
same as before. old code had `PrintfCallback printfmt = (PrintfCallback)printf;` and cb_printf was called only if set. This new code changes behaviour.